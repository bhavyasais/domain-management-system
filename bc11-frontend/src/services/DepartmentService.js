import API from '../api';

const DepartmentService = {
  getAllDepartments: async () => {
    const { data } = await API.get('departments');
    return data;
  },
  addDepartment: async (params) => {
    const { data } = await API.post('departments', params);
    return data;
  },
};

export default DepartmentService;
