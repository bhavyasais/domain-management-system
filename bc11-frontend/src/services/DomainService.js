import API from '../api';

const DomainService = {
  getAllDomainsForDepartment: async (params) => {
    const { data } = await API.get('domains', { params });
    return data;
  },
  addNewDomain: async (params) => {
    const { data } = await API.post('domains', params);
    return data;
  },
  UpdateDomainTrustGroup: async (params) => {
    const { data } = await API.put('domainTrustGroup', params);
    return data;
  },
  updateDomain: async (params) => {
    const { data } = await API.put('domains', params);
    return data;
  },
};

export default DomainService;
