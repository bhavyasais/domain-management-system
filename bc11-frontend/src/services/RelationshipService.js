import API from "../api";

const RelationshipService = {
  getAllRelations: async () => {
    const { data } = await API.get("domains");
    return data;
  },
};

export default RelationshipService;