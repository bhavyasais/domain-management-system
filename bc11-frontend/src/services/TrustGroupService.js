import API from '../api';

const TrustGroupService = {
  getAllTrustGroups: async () => {
    const { data } = await API.get('trust-groups');
    return data;
  },
  getAllTrustGroupsWithPermissions: async (params) => {
    const { data } = await API.get('permission-grant', { params });
    return data;
  },
  addNewTrustGroup: async (params) => {
    const { data } = await API.post('trust-groups', params);
    return data;
  },
  updateTrustGroupAccess: async (params) => {
    const { data } = await API.put('permission-grant', params);
    return data;
  },
  updateTrustGroup: async (params) => {
    const { data } = await API.put('trust-groups', params);
    return data;
  },
};

export default TrustGroupService;
