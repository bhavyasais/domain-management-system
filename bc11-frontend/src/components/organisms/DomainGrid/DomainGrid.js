import React, { useEffect, useImperativeHandle, useState } from 'react';
import { Grid, withStyles } from '@material-ui/core';
import { AgGridReact, AgGridColumn, AllCommunityModules } from 'ag-grid-react';
import 'ag-grid-community/dist/styles/ag-grid.css';
import 'ag-grid-community/dist/styles/ag-theme-alpine.css';
import 'ag-grid-enterprise';
import AddIcon from '@material-ui/icons/Add';
import clsx from 'clsx';
import Chip from '../../atoms/Chip/Chip';
import { createNewDomainName, getDomainCount, UpdateDomainsValue, validateDomainName } from '../../../utils/helper';
import ChipDropdown from '../ChipDropdown/ChipDropdown';
import Button from '../../atoms/Button/ButtonAtom';
import { TRUSTGROUP_COLORS } from '../../../utils/Strings';
import DomainService from '../../../services/DomainService';
import GoogleSuiteService from '../../../services/GoogleSuiteService';
const styles = () => ({
  newRowContainer: {
    height: '36px',
    width: '690',
    border: '1px solid #cccccc',
    borderTop: '0px',
    display: 'flex',
    alignItems: 'center',
    paddingLeft: '8px',
  },
  dropDownStyle: {
    width: '200px',
    marginTop: '10px',
    zIndex: '999999999 !important',
    position: 'relative',
  },
  labelStyle: {
    color: '#1d4cd7',
    textTransform: 'none',
  },
  gridContainer: {
    overflow: 'visible',
    zIndex: '0',
  },
  fullRow: {
    fontSize: '12px',
    color: '#666666',
    marginBottom: '4px',
  },
});

const DomainGrid = ({
  trustGroups,
  classes,
  setSelectedRows,
  index,
  setTotalRows,
  deptId,
  list,
  setList,
  updateDomain,
  deptName,
}) => {
  const [gridApi, setGridApi] = useState(null); // NOSONAR
  const [gridColumnApi, setGridColumnApi] = useState(null); // NOSONAR
  const [currentrow, setCurrentrow] = useState({});

  const onGridReady = (params) => {
    setGridApi(params.api);
    setGridColumnApi(params.columnApi);
  };

  const ChipRenderer = ({ value }) => {
    return (
      <Grid item xs={6}>
        <Chip label={value} backgroundColor={TRUSTGROUP_COLORS[value]} />
      </Grid>
    );
  };

  const ChipRendererGroup = ({ node }) => {
    const { key } = node;
    return (
      <Grid container direction="row" style={{ width: '250px' }} justify="space-between">
        <Grid item container direction="column" xs={6}>
          <Grid item className={classes.fullRow}>
            Group
          </Grid>

          <Grid item>
            <Chip label={key} backgroundColor={TRUSTGROUP_COLORS[key]} />
          </Grid>
        </Grid>
        <Grid item container direction="row" xs={6} alignItems="center" justify="center">
          <Grid item className={classes.fullRow}>
            Count {node.childrenAfterFilter.length}
          </Grid>
        </Grid>
      </Grid>
    );
  };

  const ChipsDropDownRenderer = React.forwardRef(({ value, data, chipsList }, ref) => {
    const [currentValue, setCurrentValue] = useState(value);
    const handleChipChange = (currentGroup) => {
      if (currentGroup) {
        setCurrentValue(currentGroup.groupName);
      }
      setCurrentrow({
        ...data,
        trustGroupName: currentGroup.groupName,
        trust_groups_id: currentGroup.id,
      });
    };

    useImperativeHandle(ref, () => ({
      getValue: () => {
        return currentValue;
      },
      isPopup: () => {
        return true;
      },
    }));

    return (
      <ChipDropdown
        list={chipsList}
        selectedValue={value}
        onDropDownChange={handleChipChange}
        ref={ref}
        customStyle={classes.dropDownStyle}
      />
    );
  });

  const getRowHeight = (params) => {
    if (params.data) {
      return 40;
    }
    return 52;
  };

  function createNewRowData() {
    const newData = {
      domain_name: createNewDomainName(list),
      trust_score: 0,
      address: 'default Address',
      relationship: 'Customer',
      created_by: 'Test User',
      trust_groups_id: trustGroups?.[0]?.id,
      departments_id: deptId,
      collaborators_id: localStorage.getItem('Collaboration'),
    };
    return newData;
  }

  const addNewDomain = async (params) => {
    try {
      const result = await DomainService.addNewDomain(params);
      gridApi.applyTransaction({
        add: [{ result }],
      });
      setList((prevData) => [...prevData, result]);
    } catch (error) {
      console.warn('error in adding domain: ', error);
    }
  };
  const addItems = () => {
    const newItems = createNewRowData();
    addNewDomain(newItems);
  };

  useEffect(() => {
    setTotalRows((prevArray) => {
      const newArray = [...prevArray];
      newArray[index] = list.length;
      return newArray;
    });
  }, [list, index]);


  const onSelectionChanged = () => {
    const selected = gridApi.getSelectedRows().length;
    setSelectedRows((prevArrray) => {
      const newArray = [...prevArrray];
      newArray[index] = selected;
      return newArray;
    });
  };

  const changeTrustGroup = async () => {
    if (currentrow) {
      const params = {
        id: currentrow.domainTrustGroupId,
        collaborators_id: currentrow.collaborators_id,
        departments_id: currentrow.departments_id,
        trust_groups_id: currentrow.trust_groups_id,
      };
      const googleObj = {
        departmentName: deptName,
        clients: [currentrow.domain_name],
        type: 'domain',
        permissionType: localStorage.getItem(currentrow.trustGroupName),
      };
      try {
        await DomainService.UpdateDomainTrustGroup(params);
        const res = await GoogleSuiteService.changePermission(googleObj);
        console.log('res of google service', res);
        const updateData = UpdateDomainsValue([...list], currentrow);
        gridApi.setRowData(updateData);
        setList(() => [...updateData]);
        setCurrentrow(null);
      } catch (error) {
        console.warn('Error in changing TrustGroup');
      }
    }
  };

  const handleCellEditingStopped = (node) => {
    if(node.newValue===node.oldValue){
      var count =getDomainCount(list,node)
      if(count>1){
      validateDomainName(list,node,gridApi,gridColumnApi)
      }
    }
    if(node.oldValue !== node.newValue){
    validateDomainName(list,node,gridApi,gridColumnApi)
    }
    if (node.column.colId === 'trustGroupName') {
      changeTrustGroup();
    } else {
      updateDomain(node.data);
    }
  };

  const totalHeigh = list.length * 40 + 40 * trustGroups.length + 251;

  return (
    <div style={{ width: '100%', height: '100%' }}>
      <div
        style={{ height: totalHeigh, width: '700px' }}
        className={clsx('ag-theme-alpine', classes.gridContainer)}
      >
        <AgGridReact
          defaultColDef={{
            resizable: true,
            sortable: true,
            filter: true,
            zIndex: 0,
            overflow: 'visible',
            editable: true,
          }}
          rowSelection="multiple"
          suppressRowTransform
          getRowHeight={getRowHeight}
          onGridReady={onGridReady}
          rowData={[...list]}
          modules={AllCommunityModules}
          groupUseEntireRow
          groupDefaultExpanded={-1}
          frameworkComponents={{ groupRowInnerRenderer: ChipRendererGroup }}
          groupRowRendererParams={{
            innerRenderer: 'groupRowInnerRenderer',
            suppressCount: true,
          }}
          onCellEditingStopped={handleCellEditingStopped}
          onSelectionChanged={onSelectionChanged}
          onSortChanged={() => {
            gridApi.refreshCells();
          }}
          rowDragManaged
        >
          <AgGridColumn
            headerName="S no"
            valueGetter="node.rowIndex + 1"
            width={78}
            lockPosition="true"
            rowDrag
          />
          <AgGridColumn headerName="Domain" field="domain_name" />
          <AgGridColumn headerName="Trust Score" field="trust_score" />
          <AgGridColumn
            headerName="Trust Group"
            field="trustGroupName"
            cellRendererFramework={ChipRenderer}
            editable
            cellEditorFramework={ChipsDropDownRenderer}
            cellEditorParams={{
              chipsList: trustGroups,
            }}
            cellStyle={{
              overflow: 'visible',
            }}
            rowGroup
          />
        </AgGridReact>
        <div className={classes.newRowContainer}>
          <Button
            variant="text"
            className={classes.button}
            startIcon={<AddIcon />}
            disableRipple
            value="New row"
            labelStyle={classes.labelStyle}
            onClick={addItems}
            textVariant="caption"
          />
        </div>
      </div>
    </div>
  );
};

DomainGrid.propTypes = {};
DomainGrid.defaultProps = { onDataChange: () => {} };

export default withStyles(styles)(DomainGrid);
