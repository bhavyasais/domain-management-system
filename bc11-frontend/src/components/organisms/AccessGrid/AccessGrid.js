import React, { useCallback, useEffect, useState } from 'react';
import { withStyles } from '@material-ui/core';
import { AgGridReact, AgGridColumn } from 'ag-grid-react';
import PropTypes from 'prop-types';
import AddIcon from '@material-ui/icons/Add';
import Button from '../../atoms/Button/ButtonAtom';

import 'ag-grid-community/dist/styles/ag-grid.css';
import 'ag-grid-community/dist/styles/ag-theme-alpine.css';
import TrustGroupService from '../../../services/TrustGroupService';
import {
  CreateTrustGroupPermissionArray,
  createUpdateTrustGroupParam,
  getPermission,
} from '../../../utils/helper';
import DomainService from '../../../services/DomainService';
import GoogleSuiteService from '../../../services/GoogleSuiteService';

const styles = () => ({
  addNewRow: {
    height: '36px',
    width: '690',
    border: '1px solid #cccccc',
    borderTop: '0',
    display: 'flex',
    alignItems: 'center',
    paddingLeft: '8px',
    backgroundColor: '#ffffff',
  },
  buttonLabelStyle: {
    color: '#1d4cd7',
    textTransform: 'none',
  },
});

const AccessGrid = ({
  classes,
  setSelectedRows,
  index,
  setTotalRows,
  trustGroups,
  deptId,
  addTrustGroup,
  updateTrustGroup,
}) => {
  const [gridApi, setGridApi] = useState(null); // NOSONAR
  const [gridColumnApi, setGridColumnApi] = useState(null); // NOSONAR
  const [list, setList] = useState([]);

  const totalHeigh = list.length * 40 + 51.4;

  const onGridReady = (params) => {
    setGridApi(params.api);
    setGridColumnApi(params.columnApi);
  };

  const updateAccess = async (item, colId, checked) => {
    const params = createUpdateTrustGroupParam(item, colId, checked);

    params.collaborators_id = localStorage.getItem('Collaboration');
    params.departments_id = JSON.parse(localStorage.getItem('dept')).id;

    const googleObj = {
      departmentName: JSON.parse(localStorage.getItem('dept')).name,
      type: 'domain',
      permissionType: localStorage.getItem(item.trustGroupName),
    };

    try {
      await TrustGroupService.updateTrustGroupAccess(params);

      const domains = await DomainService.getAllDomainsForDepartment({
        collaborators_id: localStorage.getItem('Collaboration'),
        departments_id: JSON.parse(localStorage.getItem('dept')).id,
      });

      googleObj.clients = domains
        .filter((domainObj) => domainObj.trustGroupName === item.trustGroupName)
        .map((obj) => obj.domain_name);
      console.log('google,', googleObj);

      await GoogleSuiteService.changePermission(googleObj);
    } catch (error) {
      console.warn('Error in updating trustgroup Access : ', error);
    }
  };

  const clickedCheckbox = (e, params) => {
    const { checked } = e.target;
    const { colId } = params.column;
    params.node.setDataValue(colId, checked);
    const newAarray = [...params.list];
    newAarray[params.rowIndex] = params.data;
    getPermission(newAarray);
    setList(newAarray);
    updateAccess(params.data, colId, checked);
  };

  const checkboxRenderer = (params) => {
    return (
      <div>
        <input type="checkbox" onClick={(e) => clickedCheckbox(e, params)} checked={params.value} />
      </div>
    );
  };

  const headerRenderer = (params) => {
    const name = params.displayName;
    return (
      <div>
        <input type="checkbox" onClick={() => clickedCheckbox(params)} />
        <label>{name}</label>
      </div>
    );
  };

  const fetchAllPermissions = useCallback(async () => {
    try {
      const result = await TrustGroupService.getAllTrustGroupsWithPermissions({
        collaborators_id: localStorage.getItem('Collaboration'),
        departments_id: deptId,
      });
      const permissionArray = CreateTrustGroupPermissionArray(result, trustGroups); // This should ideally be handled by backend
      setList([...permissionArray]);
    } catch (error) {
      console.warn('Error in fetch Access permissions : ', error);
    }
  }, [trustGroups, deptId]);

  useEffect(() => {
    fetchAllPermissions();
  }, [fetchAllPermissions]);

  useEffect(() => {
    gridApi?.refreshCells();
  }, [list]);

  useEffect(() => {
    setTotalRows((prevArray) => {
      const newArray = [...prevArray];
      newArray[index] = list.length;
      return newArray;
    });
  }, [list, index]);

  const handleOnSelectionChanged = () => {
    const selected = gridApi.getSelectedRows().length;
    setSelectedRows((prevArrray) => {
      const newArray = [...prevArrray];
      newArray[index] = selected;
      return newArray;
    });
  };

  function createNewTrustGroup() {
    let groupName = 'Top 10';
    while (trustGroups.find((item) => item.groupName === groupName)) {
      groupName += '0';
    }
    return groupName;
  }

  const addNewTrustGroup = () => {
    const newItems = createNewTrustGroup();
    addTrustGroup(newItems);
  };

  const handleCellEditingStopped = (param) => {
    if (param.oldValue !== param.newValue) {
      updateTrustGroup(param.data);
    }
  };

  return (
    <div style={{ width: '100%', height: '100%', flex: 1 }}>
      <div
        style={{
          height: list?.length !== 0 ? totalHeigh : '360px',
          width: '1091px',
        }}
        className="ag-theme-alpine"
      >
        <AgGridReact
          defaultColDef={{
            resizable: true,
            sortable: true,
            filter: true,
          }}
          rowSelection="multiple"
          rowDragManaged
          getRowHeight={() => 40}
          onGridReady={onGridReady}
          rowData={list}
          onSelectionChanged={handleOnSelectionChanged}
          onSortChanged={() => {
            gridApi.refreshCells();
          }}
          onRowDragEnd={() => {
            gridApi.refreshCells();
          }}
          onCellEditingStopped={handleCellEditingStopped}
        >
          <AgGridColumn
            headerName="S no"
            valueGetter="node.rowIndex + 1"
            width={78}
            lockPosition="true"
            rowDrag
          />
          <AgGridColumn headerName="A Trust Group" field="trustGroupName" editable />
          <AgGridColumn
            headerName="Edit File?"
            field="editFile"
            cellRendererFramework={checkboxRenderer}
            cellRendererParams={{
              list,
            }}
            headerComponentFramework={headerRenderer}
          />
          <AgGridColumn
            headerName="View File?"
            field="viewFile"
            cellRendererFramework={checkboxRenderer}
            cellRendererParams={{
              list,
            }}
            headerComponentFramework={headerRenderer}
          />
          <AgGridColumn
            headerName="Comment on File?"
            field="commentFile"
            cellRendererFramework={checkboxRenderer}
            cellRendererParams={{
              list,
            }}
            headerComponentFramework={headerRenderer}
          />
          <AgGridColumn
            headerName="Transfer ownership of File?"
            field="transferFile"
            cellRendererFramework={checkboxRenderer}
            cellRendererParams={{
              list,
            }}
            headerComponentFramework={headerRenderer}
          />
        </AgGridReact>
        <div className={classes.addNewRow}>
          <Button
            variant="text"
            className={classes.button}
            startIcon={<AddIcon />}
            disableRipple
            value="New row"
            labelStyle={classes.buttonLabelStyle}
            onClick={addNewTrustGroup}
            textVariant="caption"
          />
        </div>
      </div>
    </div>
  );
};

AccessGrid.propTypes = {
  setSelectedRows: PropTypes.func,
  index: PropTypes.number.isRequired,
  setTotalRows: PropTypes.func,
  trustGroups: PropTypes.arrayOf(PropTypes.string),
};
AccessGrid.defaultProps = {
  setSelectedRows: () => {},
  setTotalRows: () => {},
  trustGroups: [],
};

export default withStyles(styles)(AccessGrid);
