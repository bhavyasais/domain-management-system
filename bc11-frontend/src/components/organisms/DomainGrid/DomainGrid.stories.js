import React from "react";
import { DomainrowData } from "../../../utils/commandCentermockData";
import DomainGrid from "./DomainGrid";

export default {
  title: "Organisms/DomainGrid",
};

const trustGroups = ["Top 100", "Top 1000", "Top 3000", "Do not trust"];

export const domainGroup = () => {
  return <DomainGrid rowData={DomainrowData} trustGroups={trustGroups} />;
};
