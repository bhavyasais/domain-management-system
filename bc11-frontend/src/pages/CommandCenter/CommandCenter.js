import { withStyles } from '@material-ui/core';
import React, { useCallback, useEffect, useState } from 'react';
import AccessGrid from '../../components/organisms/AccessGrid/AccessGrid';
import DomainGrid from '../../components/organisms/DomainGrid/DomainGrid';
import RelationshipGrid from '../../components/organisms/RelationshipGrid/RelationshipGrid';
import AccessPanel from '../../components/templates/AccessPanel/AccessPanel';
import Tabs from '../../components/templates/Tabs/Tabs';
import DepartmentService from '../../services/DepartmentService';
import DomainService from '../../services/DomainService';
import TrustGroupService from '../../services/TrustGroupService';
import { trustGrouprowData } from '../../utils/commandCentermockData';
import { UpdateDomainsValue } from '../../utils/helper';
import { RELATIONS } from '../../utils/Strings';

const styles = () => ({});

const CommandCenter = () => {
  const [dept, setDept] = useState([]);
  const [currentDept, setCurrentDept] = useState(null);
  const [trustGroups, setTrustGroups] = useState([]);
  const [tabLabels, setTabLabels] = useState(['Box communication base', 'External partners']);
  const [list, setList] = useState([]);

  const onDataChange = (updatedAarray) => {
  };

  const setCurrentDepartment = (department) => {
    setCurrentDept(department);
    localStorage.setItem('dept', JSON.stringify(department));
  };

  const addDepartment = async (newDept) => {
    try {
      const result = await DepartmentService.addDepartment(newDept);
      setDept((prevDept) => [...prevDept, result]);
    } catch (err) {
      console.warn('error in adding department');
    }
  };

  const addNewTrustGroup = async (trustGroup) => {
    console.log('trsutGroup', trustGroup);
    const alreadyExist = trustGroups.find((item) => item.groupName === trustGroup);
    if (!alreadyExist) {
      try {
        const result = await TrustGroupService.addNewTrustGroup({
          groupName: trustGroup,
          description: `${trustGroup} Group`,
          username: 'test user',
        });
        setTrustGroups((prevTrustGroups) => [...prevTrustGroups, result]);
        setTabLabels((prevtabLabels) => [...prevtabLabels, trustGroup]);
      } catch (error) {
        console.warn('Error in adding new trust group : ', error);
      }
    } else {
      alert('Trustgroup already exists!');
    }
  };

  const fetchDepartments = useCallback(async () => {
    try {
      const result = await DepartmentService.getAllDepartments();
      setDept(result);
      setCurrentDept(result?.[0]);
      localStorage.setItem('dept', JSON.stringify(result[0]));
    } catch (error) {
      console.warn('error in fetching departments : ', error);
    }
  }, []);

  const fetchTrustGroups = useCallback(async () => {
    try {
      const result = await TrustGroupService.getAllTrustGroups();
      setTrustGroups(result);
      setTabLabels((prevLabels) => [...prevLabels, ...result.map((item) => item.groupName)]);
    } catch (error) {
      console.warn('error in fetching departments : ', error);
    }
  }, []);

  const fetchDomains = useCallback(async () => {
    try {
      const result = await DomainService.getAllDomainsForDepartment({
        collaborators_id: localStorage.getItem('Collaboration'),
        departments_id: currentDept?.id,
      });
      setList(result);
    } catch (error) {
      console.warn('error in fetching domains : ', error);
    }
  }, [currentDept]);

  const updateTrustGroup = async (data) => {
    const param = {
      id: data.trust_groups_id,
      description: 'trust group updated',
      groupName: data.trustGroupName,
      username: 'test user',
    };
    try {
      await TrustGroupService.updateTrustGroup(param);
    } catch (err) {
      console.warn('Error in updating trust group name : ', err);
    }
  };

  useEffect(() => {
    fetchTrustGroups();
    fetchDepartments();
  }, []);

  useEffect(() => {
    fetchDomains();
  }, [fetchDomains]);

  const updateDomain = async (data) => {
    const params = {
      id: data.id,
      trust_score: data.trust_score,
      relationship: data.relationship,
      domain_name: data.domain_name,
      username: data.created_by,
    };
    try {
      await DomainService.updateDomain(params);

      const updateData = UpdateDomainsValue([...list], data);
      setList(updateData);
    } catch (error) {
      console.warn('Error in updating domain : ', error);
    }
  };

  return (
    <div>
      <AccessPanel headerVariant="h3">
        <Tabs
          tabLabels={tabLabels}
          dept={dept}
          addDepartment={addDepartment}
          currentDept={currentDept}
          setCurrentDepartment={setCurrentDepartment}
          addTrustGroup={addNewTrustGroup}
        >
          {tabLabels.map((label, idx) => {
            if (label === 'Box communication base') {
              return (
                <AccessGrid
                  rowData={trustGrouprowData}
                  trustGroups={trustGroups}
                  deptId={currentDept?.id}
                  index={idx}
                  addTrustGroup={addNewTrustGroup}
                  updateTrustGroup={updateTrustGroup}
                  key
                />
              );
            }
            if (label === 'External partners') {
              return (
                <DomainGrid
                  trustGroups={trustGroups}
                  onDataChange={onDataChange}
                  index={idx}
                  deptId={currentDept?.id}
                  deptName={currentDept?.name}
                  key
                  list={list}
                  setList={setList}
                  updateDomain={updateDomain}
                />
              );
            }
            return (
              <RelationshipGrid
                index={idx}
                relationships={RELATIONS}
                deptId={currentDept?.id}
                trustGroup={trustGroups.find((item) => item.groupName === label)}
                key
                list={list}
                setList={setList}
                updateDomain={updateDomain}
              />
            );
          })}
        </Tabs>
      </AccessPanel>
    </div>
  );
};

CommandCenter.propTypes = {};
CommandCenter.defaultProps = {};

export default withStyles(styles)(CommandCenter);
