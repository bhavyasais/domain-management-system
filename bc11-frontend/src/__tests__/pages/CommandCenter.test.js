import React from 'react';
import renderer from 'react-test-renderer';
import CommandCenter from '../../pages/CommandCenter/CommandCenter';
import DepartmentService from '../../services/DepartmentService';
import TrustGroupService from '../../services/TrustGroupService';
import RelationshipService from '../../services/RelationshipService';
import DomainService from '../../services/DomainService';
import { mount, shallow } from 'enzyme';
import { act } from 'react-dom/test-utils';
import AccessGrid from '../../components/organisms/AccessGrid/AccessGrid';
import DomainGrid from '../../components/organisms/DomainGrid/DomainGrid';

jest.mock('../../services/DepartmentService');
jest.mock('../../services/TrustGroupService');
jest.mock('../../services/RelationshipService');
jest.mock('../../services/DomainService');
const DEPARTMENTS_RESPONSE = [
  {
    id: '51ce0b20-fd9c-41af-85a6-1e063aecbeb0',
    name: 'Sales',
  },
  {
    id: 'c1faba5e-9d82-45a3-89a0-faa21f207a6a',
    name: 'Legal',
  },
  {
    id: 'fb89416f-1f05-41d1-87f7-d908e0a3199c',
    name: 'Finance',
  },
  {
    id: 'dc7cd529-bc0f-4c1c-8b1d-7d27e75d1a35',
    name: 'Investor Relations',
  },
];

const TRUSTGROUP_RESPONSE = [
  {
    id: '228afde7-a65b-407d-b11e-e2c290559edd',
    name: 'Do not trust',
  },
  {
    id: 'd78007b4-70be-4ef3-be62-e5505e9ba3be',
    name: 'Top 100',
  },
  {
    id: 'ed7f065e-911d-4bd8-84df-cd85427b1b60',
    name: 'Top 10000',
  },
  {
    id: 'be4efe99-33a0-4de2-af43-b16db6af9782',
    name: 'Top 30000',
  },
];

const RELATIONSHIP_RESPONSE = [
  {
    id: 'c4916255-fe5c-40de-93dc-5e6b57fa1dc4',
    name: 'Zemosolabs',
    address: 'zemosolabs.com',
    trust_score: '222',
    relationship: 'Partner',
  },
  {
    id: 'd1eae897-4fa3-49f7-9071-27136b670f40',
    name: 'Ycombinator',
    address: 'ycombinator.com',
    trust_score: '222',
    relationship: 'Customer',
  },
  {
    id: '0aad7176-59be-496a-8b57-71915a9cd010',
    name: 'Clearedin',
    address: 'clearedin.com',
    trust_score: '222',
    relationship: 'Vendor',
  },
];

describe('CommandCenter', () => {
  const handleClick = jest.fn();
  it('should render correctly', () => {
    let wrapper;
    wrapper = shallow(<CommandCenter />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Service calls works', async () => {
    DepartmentService.getAllDepartments.mockReturnValue(DEPARTMENTS_RESPONSE);
    TrustGroupService.getAllTrustGroups.mockReturnValue(TRUSTGROUP_RESPONSE);
    RelationshipService.getAllRelations.mockReturnValue(RELATIONSHIP_RESPONSE);
    DomainService.getAllDomainsForDepartment.mockReturnValue(RELATIONSHIP_RESPONSE);
    let mounted;
    act(() => {
      mounted = mount(<CommandCenter />);
    });
    await act(async () => {
      await Promise.resolve(mounted);
      await Promise.resolve(mounted);
      await Promise.resolve(mounted);
      await Promise.resolve(mounted);
      mounted.update();
    });
    expect(mounted).toBeDefined();
  });
  it('testing tabs', () => {
    const wrapper = mount(<CommandCenter setCurrentDepartment={handleClick} addDepartment={handleClick}/>);
    wrapper.find('Tabs').at(0).props().setCurrentDepartment();
    wrapper.find('Tabs').at(0).props().addDepartment();
    wrapper.find('Tabs').at(0).props().addTrustGroup();
    expect(wrapper).toMatchSnapshot();
  });

  it('testing update domain', () => {
    const wrapper = mount(<CommandCenter updateDomain={handleClick}/>);
    DomainService.updateDomain({})
    wrapper.find(DomainGrid).props().updateDomain({data:{domain_name:"Top 100",trust_score:222,relationship:"Customer",username:"Bhavya"}});
    wrapper.find('DomainGrid').at(0).props().updateDomain();
    expect(wrapper).toMatchSnapshot();
  });
  
  it('testing update tg', () => {
    const wrapper = mount(<CommandCenter updateDomain={handleClick}/>);
    TrustGroupService.updateTrustGroup({})
    wrapper.find(AccessGrid).props().updateTrustGroup({data:{trustGroupName:"Top 100"}});
  });
});
