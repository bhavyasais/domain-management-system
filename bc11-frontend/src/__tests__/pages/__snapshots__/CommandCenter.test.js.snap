// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommandCenter should render correctly 1`] = `ShallowWrapper {}`;

exports[`CommandCenter testing tabs 1`] = `ReactWrapper {}`;

exports[`CommandCenter testing update domain 1`] = `ReactWrapper {}`;
